File: utilities/src/main/kotlin/lunalauf/rms/utilities/network/util/RepetitionHandler.java
package lunalauf.rms.utilities.network.util;

public class RepetitionHandler {

    private final long repetitionTimeThreshold;
    private long lastId = -1;
    private long lastTimeMillis = 0;

    public RepetitionHandler(long repetitionTimeThreshold) {
        this.repetitionTimeThreshold = repetitionTimeThreshold;
    }

    public boolean isUnwantedRepetition(long id) {
        if (id == lastId)
            if (System.currentTimeMillis() - lastTimeMillis < repetitionTimeThreshold)
                return true;
        lastId = id;
        lastTimeMillis = System.currentTimeMillis();
        return false;
    }

}

File: utilities/src/main/kotlin/lunalauf/rms/utilities/network/util/RepetitionHandler.kt
package lunalauf.rms.utilities.network.util

class RepetitionHandler(
    private val repetitionTimeThreshold: Long
) {
    private var lastId: Long = -1
    private var lastTimeMillis: Long = 0

    fun isUnwantedRepetition(id: Long): Boolean {
        if (id == lastId)
            if (System.currentTimeMillis() - lastTimeMillis < repetitionTimeThreshold)
                return true
        lastId = id
        lastTimeMillis = System.currentTimeMillis()
        return false
    }
}
